[Windows] Fixed Shadow not updated when Clipping a View with a shadow

NirmalKumarYuvaraj opened the pull request:
### Issue Details:
When applying clipping at runtime, the shadow effect fails to render properly for the view. The shadow property, which should adjust based on the clipping bounds, remains unchanged leading to incorrect shadow rendering.

### Root Cause:
The clipping update operation does not trigger a notification to the shadow property that depends on it. This missing dependency notification results in the shadow not being recalculated and updated when the clip bounds change.

### Description of Change:
Added a call to `UpdateShadowAsync()` method after clip updates to ensure the shadow property is properly recalculated. This ensures that when clipping bounds are modified at runtime, the shadow effect is updated accordingly to maintain visual consistency.

Validated the behaviour in the following platforms

- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac

### Issues Fixed

Fixes #27730


### Output

| Before | After |
| ------ | ----- |
| <video src="https://github.com/user-attachments/assets/5be5ca26-453e-441c-89b4-22e49dfb6869">|<video src="https://github.com/user-attachments/assets/df88afc0-3c92-48a7-8016-a17d2e3a562b">|

NirmalKumarYuvaraj committed:
Fixed Shadow not updated when Clipping a View with a shadow

NirmalKumarYuvaraj committed:
FIxed shadow not updating with vlip

NirmalKumarYuvaraj committed:
Updated code changes

jsuarezruiz commented on src/Core/src/Platform/Windows/WrapperView.cs:
```diff
@@ -121,6 +121,8 @@ void UpdateClip()
 
 			visual.Clip = geometricClip;
+			//When the clip is updated, the shadow must be updated as well
+			UpdateShadowAsync().FireAndForget(IPlatformApplication.Current?.Services?.CreateLogger(nameof(WrapperView)));
```
This works if the View already have a Shadow, but, works if the order is:
1. Clip
2. Add a Shadow
?

NirmalKumarYuvaraj replied:
@jsuarezruiz , A new shadow will be created if the order is reversed.

jsuarezruiz reviewed (changes requested):
Could the sample tests the two possible scenarios:

1. Have a View with a Shadow and then clip it.
2. Have a View without a Shadow, clip it and then add the shadow.

In both cases, the shadow must adapt to the clipped path.

NirmalKumarYuvaraj committed:
Updated testcase image

NirmalKumarYuvaraj committed:
Updated test case

NirmalKumarYuvaraj committed:
updated test case image

NirmalKumarYuvaraj commented:
> Could the sample tests the two possible scenarios:
> 
> 1. Have a View with a Shadow and then clip it.
> 2. Have a View without a Shadow, clip it and then add the shadow.
> 
> In both cases, the shadow must adapt to the clipped path.

@jsuarezruiz , i have updated the test case to ensure the above mentioned scenario's. I have updated the before and after video reference in the PR template. Please let me know if you have any concerns.

jsuarezruiz commented:
/azp run

jsuarezruiz commented:
/rebase

jsuarezruiz commented:
/azp run

NirmalKumarYuvaraj committed:
Added pending snapshots

jsuarezruiz commented:
/azp run
